File: packages/crosswake-shell-core-android/src/main/java/dev/crosswake/shell/core/CrosswakeShellConfig.kt
package dev.crosswake.shell.core

data class CrosswakeShellConfig(
    val appInfoDelegate: AppInfoDelegate? = null,
    val hapticsDelegate: HapticsDelegate? = null,
    val permissionStatusDelegate: PermissionStatusDelegate? = null,
    val notificationTokenDelegate: NotificationTokenDelegate? = null,
    val shareDelegate: ShareDelegate? = null,
    val filesPickDelegate: FilesPickDelegate? = null,
    val routeDelegate: RouteDelegate? = null
) {
    val registeredCapabilities: List<String>
        get() {
            val caps = mutableListOf<String>()
            if (appInfoDelegate != null) caps.add("app.info.get")
            if (hapticsDelegate != null) caps.add("haptics.impact")
            if (permissionStatusDelegate != null) caps.add("permissions.status")
            if (notificationTokenDelegate != null) caps.add("notification_token")
            if (shareDelegate != null) caps.add("share.invoke")
            if (filesPickDelegate != null) caps.add("file_picker")
            routeDelegate?.let { delegate ->
                caps.addAll(delegate.registeredRoutes.map { "route.$it" })
            }
            return caps
        }
}

File: packages/crosswake-shell-core-ios/Sources/CrosswakeShellCore/CrosswakeShellConfig.swift
import Foundation

public struct CrosswakeShellConfig {
    public let packProvider: (any PackProvider)?
    public weak var appInfoDelegate: AppInfoDelegate?
    public weak var hapticsDelegate: HapticsDelegate?
    public weak var permissionStatusDelegate: PermissionStatusDelegate?
    public weak var notificationTokenDelegate: NotificationTokenDelegate?
    public weak var shareDelegate: ShareDelegate?
    public weak var filesPickDelegate: FilesPickDelegate?
    public weak var routeDelegate: RouteDelegate?

    public init(
        packProvider: (any PackProvider)? = nil,
        appInfoDelegate: AppInfoDelegate? = nil,
        hapticsDelegate: HapticsDelegate? = nil,
        permissionStatusDelegate: PermissionStatusDelegate? = nil,
        notificationTokenDelegate: NotificationTokenDelegate? = nil,
        shareDelegate: ShareDelegate? = nil,
        filesPickDelegate: FilesPickDelegate? = nil,
        routeDelegate: RouteDelegate? = nil
    ) {
        self.packProvider = packProvider
        self.appInfoDelegate = appInfoDelegate
        self.hapticsDelegate = hapticsDelegate
        self.permissionStatusDelegate = permissionStatusDelegate
        self.notificationTokenDelegate = notificationTokenDelegate
        self.shareDelegate = shareDelegate
        self.filesPickDelegate = filesPickDelegate
        self.routeDelegate = routeDelegate
    }

    public var registeredCapabilities: [String] {
        var caps: [String] = []
        if appInfoDelegate != nil { caps.append("app.info.get") }
        if hapticsDelegate != nil { caps.append("haptics.impact") }
        if permissionStatusDelegate != nil { caps.append("permissions.status") }
        if notificationTokenDelegate != nil { caps.append("notification_token") }
        if shareDelegate != nil { caps.append("share.invoke") }
        if filesPickDelegate != nil { caps.append("file_picker") }
        if let routeDelegate = routeDelegate {
            caps.append(contentsOf: routeDelegate.registeredRoutes.map { "route.\($0)" })
        }
        return caps
    }
}
